Shared/AppFeature/GridFeature/GridState.swift

```diff
@@ -2,166 +2,170 @@ import SwiftUI
-        //          .map { row, column in
-        //              [row, column]
-        //          }
-        //          .map { row, column in
-        //              GridCell(row: row, column: column)
-        //          }
+  static var cellSize: CGFloat = 24.0
+  let fpsCorners: Double = 20.0
+  var fps: Double = 20
+  var cells: Matrix<GridCellState>
+  
+  init(rows: Int = 3, cols: Int = 4) {
+    self.cells = Matrix(rows, cols) { row, column in
+      GridCellState(row: row, column: column)
     }
+  }
 }
 
 enum GridAction {
-    case reset
-    case resetCells
-    case flipCorners
-    case clockWise
-    case updatedCell(GridCell)
+  case reset
+  case resetCells
+  case flipCorners
+  case clockWise
+  case updatedCell(GridCellState)
+  case cell(id: GridCellState.ID, action: GridCellAction)
+  case updateFPS(Double)
 }
 
 struct GridEnvironement {
-    struct FlipCornersID: Hashable {}
-    struct ClockWiseID: Hashable {}
-    
-    var mainQueue: AnySchedulerOf<DispatchQueue>
-    var uuid: () -> UUID
+  struct FlipCornersID: Hashable {}
+  struct ClockWiseID: Hashable {}
+  
+  var mainQueue: AnySchedulerOf<DispatchQueue>
+  var uuid: () -> UUID
+  
+  func reset(state: inout GridState) { // TODO: Mutability is allowed here?
+    state.cells = state.cells.map { GridCellState(row: $0, column: $1) }
+  }
+  
+  // Turn all cells gray, start at(0,0), turn corner cells yellow in coordinated fashion.
+  func flipCorners(state: inout GridState, fps: Int = 6) -> Effect<GridCellState, Never> {
+    reset(state: &state)
+    let publisher = state.cells.entries
+      .flatMap { $0 }
+      .compactMap { cell -> GridCellState? in
+        let cornerRow = cell.row == 0 || cell.row == state.cells.rows - 1
+        let cornerColumn = cell.column == 0 || cell.column == state.cells.cols - 1
+        return cornerRow && cornerColumn ? cell : nil
+      }
+      .publisher
+      .zip(Effect.timer(id: FlipCornersID(), every: .seconds(1.0 / Double(fps)), on: mainQueue))
+      .map { cell, _ -> GridCellState in
+        GridCellState(row: cell.row, column: cell.column, rotation: cell.rotation + 90.0, color: .yellow)
+      }
+      .eraseToAnyPublisher()
+      .eraseToEffect()
+    return publisher
+  }
+  
+  func cancelFlipCorners() -> Effect<GridAction, Never> {
+    Effect.cancel(id: FlipCornersID())
+  }
+  
+  // Turn all cells gray, start at(0,0), move in clockwise fashion, turn visited cells yellow.
+  func clockWise(state: inout GridState, fps: Int = 6) -> Effect<GridCellState, Never> {
     
-    func reset(state: inout GridState) {
-        state.cells = state.cells.map { row, column in
-            GridCell(row: row, column: column)
-        }
-    }
+    // Prepare to move.
+    var currentMove = Move(
+      minRows: 0, maxRows: state.cells.rows,
+      minColumns: 0, maxColumns: state.cells.cols,
+      row: 0, column: 0, direction: .right
+    )
+    var orderedCells = [state.cells.at(currentMove.row,currentMove.column)]
```

Comment:
will be come
```
var orderedCells = [state.cells[currentMove.row, currentMove.column]]
```

Reply:
fixed via commit: f8f2dc3f37cd562687a7451dcaf5ec8d1436a3dd